Newsletters: add 253 (2023-05-31) (#1130)
bitcoinops/bitcoinops.github.io: _posts/en/newsletters/2023-05-31-newsletter.md

      current offchain values for an equivalent amount of new offchain
      values.  This can be performed quickly because a failure of the
      onchain component will simply unwind the swap, returning all funds
      to where they started.  A blinding protocol similar to those used
      by some existing [coinjoin][topic coinjoin] implementations can
      prevent any user or the counterparty from determining which user
      ended up with which bitcoins.

    - *Making internal transfers:* one user can transfer their offchain
      funds to another user with the same counterparty.  The atomicity
      assures that either the receiver will get their money or the
      spender receives a refund.  For a receiver that doesn't trust both
      the spender and the counterparty, they will need to wait for as
      many confirmations as they would for a regular onchain
      transaction.

        Keceli and a commentator [link][keceli reply0] to
        [previous][harding reply0] research describing how a zero-conf
        payment can be made uneconomical to double spend by pairing it
        with a fidelity bond that can be claimed by any miner who

>         observed both versions of the double spent transaction.  That

Reviewer: ```suggestion
        observed both versions of the double-spent transaction.  That
```

    counterparty, although the individual transactions are fairly small.

    Comments about the protocol posted to the mailing list included:

    - *A request for more documentation:* at [least][stone reply] two
      [respondents][dryja reply] requested additional documentation
      about how the system worked, finding it hard to analyze given the
      high-level description provided to the mailing list.  Keceli has
      since begun publishing [draft specifications][arc specs].

    - *Concern that receiving is slow compared to LN:* [several][dryja
      reply] people [noted][harding reply1] that, in the initial design,
      it's not possible to trustlessly receive a payment from the
      joinpool (either offchain or onchain) without waiting for a
      sufficient number of confirmations.  That can take hours, whereas
      many LN payments currently complete in less than a second. Even
      with fidelity bonds, LN would be faster on average.

    - *Concern that the onchain footprint is high:* one [reply][jk_14]
      noted that, at one transaction every five seconds, about 200 such
      counterparties would consume the entire space of every block.

>       Another [reply][harding reply0] assumed the size of the

Reviewer: ```suggestion
      Another [reply][harding reply0] assumed the size of each of the
```

      counterparty's onchain transactions was roughly the size of an LN
      channel open or cooperative close transaction, so a counterparty
      with a million users that creates 6.3 million onchain
      transactions per year would use an equivalent amount of space to
      each of those users opening or closing an average of 6.3 channels
      each per year; thus, LN's onchain costs could be lower than using
      the counterparty until it had reached massive scale.

    - *Concern about a large hot wallet and the capital costs:* a
      [reply][harding reply0] considered that the counterparty would

>       need to keep an amount of bitcoins on hand (probably in a hot

Reviewer: ```suggestion
      need to keep an amount of bitcoin on hand (probably in a hot
```

      bitcoins back for a period of up to 28 days under the current
      design proposal.  If the counterparty charged a low interest rate
      of 1.5% per year on their capital, that would be an equivalent
      charge of 0.125% on the amount of every transaction performed with
      the involvement of the counterparty (including coinjoins, internal
      transfers, and LN payments).  By comparison, [public
      statistics][1ml stats] available at the time of writing (collected
      by 1ML) indicate a median feerate for LN transfers of 0.0026%,
      almost 50 times lower.

    Several comments on the list were also excited for the proposal and
    were looking forward to seeing Keceli and others explore the design
    space of managed joinpools.

- **Transaction relay over Nostr:** Joost Jager [posted][jager nostr] to
  the Bitcoin-Dev mailing list to request feedback on the idea by Ben
  Carman of using the [Nostr][] protocol for relaying transactions that
  might not propagate well on the P2P network of Bitcoin full nodes that
  provide relay services.

    In particular, Jager examines the possibility of using Nostr for the

>     relay of transaction packages---the ability to relay an ancestor

Reviewer 2: As this is not the only use of relaying packages
```suggestion
    relay of transaction packages---in particular, the ability to relay an ancestor
```

      charge of 0.125% on the amount of every transaction performed with
      the involvement of the counterparty (including coinjoins, internal
      transfers, and LN payments).  By comparison, [public
      statistics][1ml stats] available at the time of writing (collected
      by 1ML) indicate a median feerate for LN transfers of 0.0026%,
      almost 50 times lower.

    Several comments on the list were also excited for the proposal and
    were looking forward to seeing Keceli and others explore the design
    space of managed joinpools.

- **Transaction relay over Nostr:** Joost Jager [posted][jager nostr] to
  the Bitcoin-Dev mailing list to request feedback on the idea by Ben
  Carman of using the [Nostr][] protocol for relaying transactions that
  might not propagate well on the P2P network of Bitcoin full nodes that
  provide relay services.

    In particular, Jager examines the possibility of using Nostr for the
    relay of transaction packages---the ability to relay an ancestor
    transaction with a feerate below the minimum accepted value by
    bundling it with a descendant that pays a high enough fee to

>     compensate for its relative's deficiency.  This makes [CPFP][topic

Reviewer 3: Maybe?

```suggestion
    compensate for its parent's deficiency.  This makes [CPFP][topic
```

- [Bitcoin Core #27469][] speeds up Initial Block Download (IBD) when
  one or more wallets are being used.  With this change, a block will
  only be scanned for transactions matching a particular wallet if it
  was mined after the wallet's birthdate---the date recorded in the
  wallet as when it was created.

- [Bitcoin Core #27626][] allows a peer who has requested our node
  provide [compact block relay][topic compact block relay] in
  high-bandwidth mode to make up to three requests for transactions from
  the latest block we advertised to them.  Our node will respond to the
  request even if we didn't initially provide them with a compact block.
  This allows a peer who receives a compact block from one of its other
  peers to request any missing transactions from us, which can help if
  that other peer has become unresponsive.  This can help our peer
  validate the block faster, which may also help them use it sooner in
  time-critical functions, such as mining.

- [Bitcoin Core #25796][] adds a new `descriptorprocesspsbt` that can be
  used to update a [PSBT][topic psbt] with information that will help it
  later be signed or finalized.  A [descriptor][topic descriptors]

>   provided to the RPC will be used to retrieve information from mempool

Reviewer: ```suggestion
  provided to the RPC will be used to retrieve information from the mempool
```

  an outdated state, discouraging them from attempting theft.  However,
  if the remote peer accepts an HTLC that will pay them if it's
  successful, they will have more at stake than the reserve anyway.  If
  it's not successful, their balance will return to the previous amount,
  which will have been above the reserve.

    This is a mitigation for a *stuck funds problem*, which occurs when
    a payment would cause the party responsible for paying the fees to
    need to pay more value than their current available balance, even
    when they might be the party receiving the payment.  For previous
    discussion of this problem, see [Newsletter #85][news85 stuck
    funds].

- [BTCPay Server 97e7e][] begins setting the [BIP78][] `minfeerate`
  (minimum feerate) parameter for [payjoin][topic payjoin] payments.
  See also the [bug report][btcpay server #4689] that lead to this
  commit.

- [BIPs #1446][] makes a small change and a number of additions to the
  [BIP340][] specification of [schnorr signatures][topic schnorr
  signatures] for Bitcoin-related protocols.  The change is allowing the
  message to be signed to be an arbitrary length; previous versions of

>   the BIP required the message be exactly 32 bytes.  See a related

Reviewer: ```suggestion
  the BIP required the message to be exactly 32 bytes.  See a related
```
or
```suggestion
 the BIP required that the message be exactly 32 bytes.  See a related
```

  change to the Libsecp256k1 library described in [Newsletter
  #157][news157 libsecp].  The change has no effect on the use of BIP340
  in consensus applications as signatures used with both [taproot][topic
  taproot] and [tapscript][topic tapscript] (respectively, BIPs
  [341][bip341] and [342][bip342]) use 32-byte messages.

    The additions describe how to effectively use arbitrary length

>     messages, recommend how to use a hash tag prefix, and provide

Reviewer: not sure, but maybe
```suggestion
    messages, recommend how to use a hashtag prefix, and provide
```
